src/coffee/import.coffee

```diff
@@ -59,31 +60,35 @@ class Import
           # - process products in batches!!
           # - for each chunk match products -> createOrUpdate
           # - provide a way to accumulate partial results, or just log them to console
-          products = []
           console.log "Mapping #{_.size rawProducts} product(s) ..."
-          for rawProduct in rawProducts
-            products.push @validator.map.mapProduct(rawProduct)
-
-          if _.size(@validator.map.errors) isnt 0
-            Promise.reject @validator.map.errors
-          else
-            console.log "Mapping done. About to process existing product(s) ..."
-            @client.productProjections.staged().all().fetch()
-            .then (payload) =>
-              existingProducts = payload.body.results
-              console.log "Comparing against #{payload.body.total} existing product(s) ..."
-              @initMatcher existingProducts
-              productsToUpdate =
-              if @validator.updateVariantsOnly
-                @mapVariantsBasedOnSKUs existingProducts, products
-              else
-                products
-              console.log "Processing #{_.size productsToUpdate} product(s) ..."
-              @createOrUpdate(productsToUpdate, @validator.types)
-            .then (result) ->
-              # TODO: resolve with a summary of the import
-              console.log "Finished processing #{_.size result} product(s)"
-              Promise.resolve result
+          # for rawProduct in rawProducts
+          #   products.push @validator.map.mapProduct(rawProduct)
+          products = rawProducts.map((p) => @validator.map.mapProduct p)
+          Promise.all(chunk(products, 100).map((ps) => @processProducts(ps)))
```

Comment:
`100` is a bit too much, mostly because of the query string max length limitation, especially if you are going to match variants by `sku`.

I would suggest 20-30 at a time.

Also you don't need to use `lodash.chunk`, there is already a function called `batchList` from `underscore-mixins`
https://github.com/sphereio/underscore-mixins#_batchlist

Another thing, you are not handling any concurrency here, you should use `Promise.map(list, fn, { concurrency: 20 })` to actual have a concurrency limit.